voyager-transitions/src/commonMain/kotlin/cafe/adriel/voyager/transitions/ScreenTransition.kt

```diff
@@ -84,6 +110,28 @@ public fun ScreenTransition(
     modifier: Modifier = Modifier,
     content: ScreenTransitionContent = { it.Content() }
 ) {
+    ScreenTransition(
+        navigator = navigator,
+        transition = transition,
+        modifier = modifier,
+        disposeScreenAfterTransitionEnd = false,
+        content = content,
+    )
+}
+
+@ExperimentalVoyagerApi
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+public fun ScreenTransition(
+    navigator: Navigator,
+    transition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform,
+    modifier: Modifier = Modifier,
+    disposeScreenAfterTransitionEnd: Boolean = false,
+    content: ScreenTransitionContent = { it.Content() }
+) {
+    // This can be costly because is checking every single item in the list
+    // we should re evaluate how validation works, maybe validating screen keys or ===
+    val previousItems = rememberPrevious(navigator.items)
```

Comment:
I think we should collect all navigator items changes. 

Case:

We have navigator with transitions, it's stack: `A -> B -> C`.
User presses "back" 2 times quickly, so the stack now: `A`. 

IMPORTANT NOTE: the transition was not completed during the entire stack change. 

After all these actions, screens B, C should be disposed. But as i understand from your solution, only screen B will be disposed

Reply:
So because of this case i collect all navigator item changes: https://github.com/adrielcafe/voyager/pull/427/files#diff-b8d3186fe9802c02964e8e565ad0ae0195d79ae7e74854c8b3c256ba68e572daR70

Reply:
So my idea is simple:

Collect all screens created during transitions. After completion of transition compare collected screens with idled stack. Then dispose screens that are no longer in the navigation stack

Reply:
Yeh, I have being testing exactly that previously and I was able to reproduce it! I will try your suggestion.